pkg/kubelet/cm/cgroup_manager_linux.go

```diff
@@ -622,3 +623,86 @@ func (m *cgroupManagerImpl) MemoryUsage(name CgroupName) (int64, error) {
 	val, err := fscommon.GetCgroupParamUint(path, file)
 	return int64(val), err
 }
+
+// Get the memory limit in bytes applied to the cgroup
+func (m *cgroupManagerImpl) GetCgroupMemoryConfig(name CgroupName) (uint64, error) {
+	cgroupPaths := m.buildCgroupPaths(name)
+	cgroupMemoryPath, found := cgroupPaths["memory"]
+	if !found {
+		return 0, fmt.Errorf("failed to build memory cgroup fs path for cgroup %v", name)
+	}
+	memLimit, err := fscommon.GetCgroupParamUint(cgroupMemoryPath, "memory.limit_in_bytes")
+	if err != nil {
+		return 0, fmt.Errorf("failed to get memory.limit_in_bytes for cgroup %v: %v", name, err)
+	}
+	return memLimit, nil
+}
+
+// Get the cpu quota, cpu period, and cpu shares applied to the cgroup
+func (m *cgroupManagerImpl) GetCgroupCpuConfig(name CgroupName) (int64, uint64, uint64, error) {
+	cgroupPaths := m.buildCgroupPaths(name)
+	cgroupCpuPath, found := cgroupPaths["cpu"]
+	if !found {
+		return 0, 0, 0, fmt.Errorf("failed to build CPU cgroup fs path for cgroup %v", name)
+	}
+	cpuQuotaStr, errQ := fscommon.GetCgroupParamString(cgroupCpuPath, "cpu.cfs_quota_us")
```

Comment:
yes, anything to do with cgroups should support v2 as well :+1:

Reply:
This is using common library. What action is needed here?

Reply:
Whole resize feature is only worked in cgroupv1 env now. But kubernetes has supported cgroupv2.

You can see the `MemoryUsage` implementation as an example. (I can do something for this problem if you need)

Reply:
Ah.. so we need to have switch. Any help on this would be very very welcome. I'm working on this during evening/night and I don't have any cycles to pick up something new and test it thoroughly for both v1/v2 cgs.

Reply:
fine, I'll try to submit a PR to this PR if this one is not merged in the next one or two weeks

Reply:
@vinaykul I think its fine to defer v2 support in this first PR.  We can say that the feature gate should not be enabled on a v2 aware host, and make v2 support a beta criteria.

Reply:
cgroupv2 support has been added and reviewed. I was not comfortable merging this without cgroupv2 once COS switched over to cgroupv2 with 1.25. IMHO this issue has been addressed. Resolving this conversation.